utils/create_dated_directory.py

```diff
@@ -0,0 +1,41 @@
+"""
+create_dated_dir Module
+
+This module provides a function for creating dated directories in the source and target directories
+based on the current date. The created directories follow a specific naming convention with the
+current date and an execution number.
+
+"""
+
+import datetime as dt
+import os
+
+
+def producer_dated_dir(source_dir: str, target_dir: str) -> tuple:
```

Comment:
this function seems more specific to our application, it is not necessary all time to create dated directory in both source and target directory , also no need for another file add this function to directory_utils package